README.rst

```diff
@@ -0,0 +1,121 @@
+    :target: https://github.com/klmitch/overcover/pulls
+.. image:: https://goreportcard.com/badge/github.com/klmitch/overcover
+    :target: https://goreportcard.com/report/github.com/klmitch/overcover
+
+This repository contains Overcover.  Overcover is a testing tool meant
+to work in conjunction with the coverage profile data generated by the
+``go test`` tool with the ``-coverprofile`` option; it computes an
+overall coverage of the entire source base and emits that
+information.  Overcover can also optionally enforce a minimum coverage
+threshold, allowing test jobs to be failed if the testing coverage has
+been weakened.
+
+Basic Usage
+===========
+
+To use Overcover, install and execute it like so::
+
+    % overcover --coverprofile coverage.out
+
+This will read the indicated coverage profile and emit a message
+describing the overall coverage represented by the profile.  A
+threshold can be enforced, which will cause Overcover to exit with a
+status code of 1 if the coverage does not reach the threshold::
+
+    % overcover --coverprofile coverage.out --threshold 75
+
+These values can also be passed to Overcover using environment
+variables named after the long option names--e.g.,
+``OVERCOVER_COVERPROFILE`` specifies a value for ``--coverprofile``,
+and ``OVERCOVER_THRESHOLD`` specifies a value for ``--threshold``.
+
+Configuration File
+==================
+
+Overcover can also read a configuration file, which may be in any
+number of formats as long as the file has a proper extension; e.g.,
+for a YAML file, the configuration file name should end with ".yaml"
+or ".yml".  The values in this file are named the same as the command
+line options, and the configuration file may be specified to Overcover
+using either ``OVERCOVER_CONFIG`` or the ``--config`` command line
+option::
+
+    % overcover --config config.yaml
+
+An example of that file could be as follows::
+
+    ---
+    coverprofile: coverage.out
+    threshold: 75
+
+Automatically Updating the Threshold
+------------------------------------
+
+When using a configuration file with Overcover, the Overcover tool can
+automatically update the threshold in the file, to help ensure that
+coverage can never decrease.  This is done using the *headroom*
+options, ``--min-headroom`` and ``--max-headroom``
+(``OVERCOVER_MIN_HEADROOM`` and ``OVERCOVER_MAX_HEADROOM`` for
+environment variables, and ``min_headroom`` and ``max_headroom`` in
+the configuration files).  If the total coverage exceeds the threshold
+plus the maximum headroom, then the threshold will be updated to be
+the total coverage minus the minimum headroom.  An example of a
+configuration file set up to use this automatic update feature would
+be as follows::
+
+    ---
+    coverprofile: coverage.out
+    threshold: 75
+    min_headroom: 1
+    max_headroom: 2
+
+Preventing Automatic Update
+---------------------------
+
+Sometimes it is convenient to prevent an automatic update of the
```

Comment:
What's the unconfigured default? 

Touching the file on disk and updating the threshold?

Have a slight pref for making the unconfigured default "touch nothing".

Reply:
Unconfigured default is "touch nothing", actually.  I'll document that :)